Code/Tools/AssetProcessor/native/tests/SourceAssetReferenceTests.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzTest/AzTest.h>
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AssetManager/SourceAssetReference.h>
+#include "UnitTestUtilities.h"
+
+namespace UnitTests
+{
+    using SourceAssetReferenceTests = ::UnitTest::ScopedAllocatorSetupFixture;
+
+    TEST_F(SourceAssetReferenceTests, ConstructBasic)
```

Comment:
Nit: It might be better to name the tests following the Osherove naming method: https://wiki.agscollab.com/display/lmbr/Test+Writing+Standards#TestWritingStandards-TheOsheroveNamingMethod

Reply:
Updated